frontend/src/utils/FileUtils.ts

```diff
@@ -0,0 +1,72 @@
+import { useLazyQuery } from "@apollo/client";
+import { saveAs } from "file-saver";
+import { File, GET_FILE } from "../APIClients/queries/FileQueries";
+
+// Retrieved from: https://stackoverflow.com/a/16245768
+const b64toBlob = (
+  b64Data: string,
+  contentType = "",
+  sliceSize = 512,
+): Blob => {
+  const byteCharacters = atob(b64Data);
+  const byteArrays = [];
+
+  for (let offset = 0; offset < byteCharacters.length; offset += sliceSize) {
+    const slice = byteCharacters.slice(offset, offset + sliceSize);
+
+    const byteNumbers = new Array(slice.length);
+    for (let i = 0; i < slice.length; i += 1) {
+      byteNumbers[i] = slice.charCodeAt(i);
+    }
+
+    const byteArray = new Uint8Array(byteNumbers);
+    byteArrays.push(byteArray);
+  }
+
+  const blob = new Blob(byteArrays, { type: contentType });
+  return blob;
+};
+
+const getMIMETypeFromExtension = (ext: string) => {
+  switch (ext) {
+    case "txt":
+      return "text/plain;charset=utf-8";
+    case "png":
+      return "image/png";
+    case "jpg":
+      return "image/jpeg";
+    case "jpeg":
+      return "image/jpeg";
+    case "doc":
+      return "application/msword";
+    case "docx":
+      return "application/vnd.openxmlformats-officedocument.wordprocessingml.document";
+    case "pdf":
+      return "application/pdf";
+    default:
+      return "application/octet-stream";
```

Comment:
is this a good catch-all/default file type? what extension does this end up as?

Reply:
oh so the extension is defined by the `file.ext` field but `application/octet-stream` serves as a default type for unknown file types:

| application/octet-stream is the default value for all other cases. An unknown file type should use this type. Browsers pay a particular care when manipulating these files, to protect users from software vulnerabilities and possible dangerous behavior.

Source: https://developer.mozilla.org/en-US/docs/Web/HTTP/Basics_of_HTTP/MIME_types/Common_types

it can also be used for `.bin` files

Reply:
looks good to me, ty for including some further reading!